model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorageProviderFactory.java

```diff
@@ -232,4 +201,31 @@ private Connection getConnection() {
             throw new RuntimeException("Failed to connect to database", e);
         }
     }
+
+    private void printOperationalInfo(Connection connection) {
+        try {
+            HashMap<String, String> operationalInfo = new LinkedHashMap<>();
+            DatabaseMetaData md = connection.getMetaData();
+            operationalInfo.put("databaseUrl", md.getURL());
+            operationalInfo.put("databaseUser", md.getUserName());
+            operationalInfo.put("databaseProduct", md.getDatabaseProductName() + " " + md.getDatabaseProductVersion());
+            operationalInfo.put("databaseDriver", md.getDriverName() + " " + md.getDriverVersion());
+
+            logger.debugf("Database info: %s", operationalInfo.toString());
+        } catch (SQLException e) {
+            logger.warn("Unable to prepare operational info due database exception: " + e.getMessage());
+        }
+    }
+
+    private void update(Class<?> modelType, Connection connection, KeycloakSession session) {
+        KeycloakModelUtils.runJobInTransaction(session.getKeycloakSessionFactory(), (KeycloakSession lockSession) -> {
+            DBLockProvider dbLock = session.getProvider(DBLockProvider.class);
+            dbLock.waitForLock(DBLockProvider.Namespace.DATABASE);
```

Comment:
The lock should be based also on model type, so that concurrent creation of tables for distinct areas is possible

Reply:
https://issues.redhat.com/browse/KEYCLOAK-19896

Reply:
Please introduce a GH issue instead